app/bundles/LeadBundle/Services/AnonymizeContactCompanyData.php

```diff
@@ -0,0 +1,378 @@
+            'companydescription' => 'description',
+            'companyindustry'    => 'industry',
+            'companyemployees'   => 'employees',
+            'companyrevenue'     => 'revenue',
+            'companystatus'      => 'status',
+            'companytype'        => 'type',
+            'companyalias'       => 'alias',
+        ];
+
+    public function __construct(
+        private readonly FieldModel $fieldModel,
+        private readonly LoggerInterface $logger,
+        private readonly EmailModel $emailModel,
+        private readonly SubmissionModel $submissionModel,
+        private readonly EntityManagerInterface $entityManager,
+        private readonly AuditLogModel $auditLogModel,
+        private readonly CompanyModel $companyModel,
+    ) {
+    }
+
+    /**
+     * @param array<Lead>|array<Company> $leadsCompanies
+     *
+     * @return array<mixed>
+     */
+    public function setHashes(array $leadsCompanies, LeadField $field, bool $pseudonymize): array
+    {
+        foreach ($leadsCompanies as $key => $leadCompany) {
+            if (!method_exists($leadCompany, 'getField')) {
+                continue;
+            }
+            $leadField = false;
+            if ($leadCompany instanceof Company && 'company' === $field->getObject()) {
+                $leadField = $leadCompany->getField($field->getAlias());
+                if (false === $leadField) {
+                    $alias          = self::COMPANY_FIELDS_TO_COLUMNS[$field->getAlias()] ?? $field->getAlias();
+                    $leadFieldValue = $leadCompany->getFieldValue($alias);
+                    if (property_exists($leadCompany, $alias) && null !== $leadFieldValue) {
+                        $tempLeadField = $leadCompany->{'get'.ucfirst($alias)}();
+                        if (empty($tempLeadField)) {
+                            continue;
+                        }
+                        unset($leadField);
+                        $leadField['value'] = $tempLeadField;
+                        $leadField['type']  = $field->getType();
+                    } else {
+                        continue;
+                    }
+                }
+            }
+            if ($leadCompany instanceof Lead && 'lead' === $field->getObject()) {
+                $leadField = $leadCompany->getField($field->getAlias());
+                if (false === $leadField) {
+                    continue;
+                }
+
+                $field     = $this->fieldModel->getRepository()->find($leadField['id']);
+
+                if (null === $field) {
+                    continue;
+                }
+            }
+            if (false === $leadField) {
+                continue;
+            }
+
+            $leadsCompanies[$key] = $this->setHash($leadCompany, $leadField, $field, $pseudonymize);
+        }
+
+        return $leadsCompanies;
+    }
+
+    /**
+     * @param array<string, string|null> $field
+     */
+    private function setHash(
```

Comment:
This block is a bit confusing due to the mix of data types and reassignments (e.g., $leadField being reused for both array and object). Consider adding a comment or splitting the logic for better readability.